cloudflare/resource_cloudflare_custom_ssl.go

```diff
@@ -0,0 +1,284 @@
+	certID := d.Id()
+	log.Printf("[DEBUG] zone ID: %s", zoneID)
+
+	zcso, err := expandToZoneCustomSSLOptions(d)
+	if err != nil {
+		return fmt.Errorf("Failed to update custom ssl cert: %s", err)
+	}
+
+	res, err := client.UpdateSSL(zoneID, certID, zcso)
+	if err != nil {
+		fmt.Printf("Failed to update custom ssl cert: %s", err)
+	}
+
+	log.Printf("[DEBUG] Custom SSL set to: %s", res.ID)
+
+	zcsp, err := expandToZoneCustomSSLPriority(d)
+	if err != nil {
+		return fmt.Errorf("Failed to update custom ssl cert: %s", err)
+	}
+
+	resList, err := client.ReprioritizeSSL(zoneID, zcsp)
+	if err != nil {
+		return fmt.Errorf("Failed to update / reprioritize custom ssl cert: %s", err)
+	}
+	log.Printf("[DEBUG] Custom SSL reprioritized to: %#v", resList)
+
+	return resourceCloudflareCustomSslRead(d, meta)
+}
+
+func resourceCloudflareCustomSslRead(d *schema.ResourceData, meta interface{}) error {
+
+	client := meta.(*cloudflare.API)
+	zoneID := d.Get("zone_id").(string)
+	certID := d.Id()
+
+	record, err := client.SSLDetails(zoneID, certID)
+	if err != nil {
+		log.Printf("[WARN] Removing record from state because it's not found in API")
+		d.SetId("")
+		return nil
+	}
+
+	d.SetId(record.ID)
+	d.Set("hosts", record.Hosts)
+	d.Set("issuer", record.Issuer)
+	d.Set("signature", record.Signature)
+	d.Set("status", record.Status)
+	d.Set("bundle_method", record.BundleMethod)
+	d.Set("uploaded_on", record.UploadedOn.Format(time.RFC3339Nano))
+	d.Set("expires_on", record.ExpiresOn.Format(time.RFC3339Nano))
+	d.Set("modified_on", record.ModifiedOn.Format(time.RFC3339Nano))
+	d.Set("priority", record.Priority)
+	return nil
+}
+
+func resourceCloudflareCustomSslDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+	zoneID := d.Get("zone_id").(string)
+	certID := d.Id()
+
+	log.Printf("[DEBUG] Deleting SSL cert %s for zone %s", certID, zoneID)
+
+	err := client.DeleteSSL(zoneID, certID)
+	if err != nil {
+		errors.Wrap(err, "failed to delete custom ssl cert setting")
+	}
+	return nil
+}
+
+func resourceCloudflareCustomSslImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	// split the id so we can lookup
+	idAttr := strings.SplitN(d.Id(), "/", 2)
+
+	if len(idAttr) != 2 {
+		return nil, fmt.Errorf("invalid id (\"%s\") specified, should be in format \"zoneID/certID\"", d.Id())
+	}
+
+	zoneID, certID := idAttr[0], idAttr[1]
+
+	log.Printf("[DEBUG] Importing Cloudflare Custom SSL Cert: id %s for zone %s", certID, zoneID)
+
+	d.Set("zone_id", zoneID)
+	d.SetId(certID)
+
+	resourceCloudflareCustomSslRead(d, meta)
+
+	return []*schema.ResourceData{d}, nil
+}
+
+func expandToZoneCustomSSLPriority(d *schema.ResourceData) ([]cloudflare.ZoneCustomSSLPriority, error) {
+	data, dataOk := d.GetOk("custom_ssl_priority")
+	log.Printf("[DEBUG] Custom SSL priority found in config: %#v", data)
+	//type mt map[string]interface{}
+	var mtSlice []cloudflare.ZoneCustomSSLPriority
+	if dataOk {
```

Comment:
Personal preference here (so take it or leave it 😛) but I'd look to return early here and instead of nesting another level as while it's not too big of a deal now, Go is pretty good at getting gnarly nested `if` expressions. I.e.

```go
data, dataOk := d.GetOk("custom_ssl_priority")

// ..

if !dataOk {
    // Perhaps a better exception than `nil` but 🤷‍♂ 
    return mtSlice, nil
}

for _, innerData := range data.([]interface{}) { 
    newData := make(map[string]interface{})
    // .. continue on
}
```

Reply:
This appears to be more idiomatic so Ill take your suggestion, thanks!